system_metrics_collector/CMakeLists.txt

```diff
@@ -28,6 +28,7 @@ endif()
 
 # find dependencies
 find_package(ament_cmake REQUIRED)
+find_package(class_loader REQUIRED)
```

Comment:
Is this needed as a build / executable dependency or just test? 

Doesn't this requirement also need to be added to package.xml?

Reply:
Hmm, it's strange that our CI managed to catch missing requirements in `package.xml` for `rclcpp_components`, but not for `class_loader`.

I'll add `class_loader` as a `test_depend` in our `package.xml`.